CHANGELOG.md

```diff
@@ -0,0 +1,755 @@
+-   [JENKINS-47109](https://issues.jenkins-ci.org/browse/JENKINS-47109) -
+    Add support for `failFast` for parallel `stage`s
+-   [JENKINS-47193](https://issues.jenkins-ci.org/browse/JENKINS-47193) -
+    Don't break parsing of `Jenkinsfile` due to `class` or `enum`
+    defined in it as well as `pipeline` block.
+
+## Version 1.2
+(Sept 21, 2017)
+
+Declarative Pipeline runs in progress upon upgrade from versions prior
+to 1.2 will probably fail on resuming.
+
+
+
+-   [JENKINS-41334](https://issues.jenkins-ci.org/browse/JENKINS-41334) -
+    Parallel definition and execution of `stage`s.
+-   [JENKINS-45198](https://issues.jenkins-ci.org/browse/JENKINS-45198) -
+    When on core 2.60+ with `pipeline-scm-step` 2.6+ and recent versions
+    of `git`, `subversion`, or `mercurial` plugins, include SCM-provided
+    variables in environment automatically.
+-   [JENKINS-42753](https://issues.jenkins-ci.org/browse/JENKINS-42753) -
+    Rewrite of runtime parser, fixing a bunch of issues
+    with `environment`, `when expression`, variable and function
+    behavior consistency, and more.
+    -   Related tickets:
+        -   [JENKINS-44298](https://issues.jenkins-ci.org/browse/JENKINS-44298) -
+            Usage of variables and functions defined outside
+            the `pipeline` block can work in `when expression` again.
+        -   [JENKINS-44482](https://issues.jenkins-ci.org/browse/JENKINS-44482) -
+            Pre-existing environment variables containing backslashes do
+            not have their backslashes discarded.
+        -   [JENKINS-44603](https://issues.jenkins-ci.org/browse/JENKINS-44603) -
+            Usage of variables and functions defined outside
+            the `pipeline` block work in `environment` variables again.
+        -   [JENKINS-45636](https://issues.jenkins-ci.org/browse/JENKINS-45636) -
+            Backslashes are correctly escaped when defined in one
+            new `environment` variable and referenced in another.
+        -   [JENKINS-45637](https://issues.jenkins-ci.org/browse/JENKINS-45637) -
+            Environment variables containing multiple pre-existing
+            environment variables no longer silently fail.
+        -   [JENKINS-45916](https://issues.jenkins-ci.org/browse/JENKINS-45916) -
+            Pre-existing environment variables, such as `PATH`, can be
+            overridden in the `environment` directive.
+        -   [JENKINS-44034](https://issues.jenkins-ci.org/browse/JENKINS-44034) -
+            Variable references in `environment` variable definitions no
+            longer require curly braces.
+        -   [JENKINS-46112](https://issues.jenkins-ci.org/browse/JENKINS-46112) -
+            Logs for `error` steps visible in Blue Ocean correctly.
+        -   [JENKINS-45991](https://issues.jenkins-ci.org/browse/JENKINS-45991) - `environment` variables
+            can be configured to have default values,
+            i.e., `FOO = FOO ?: "default"```
```

Comment:
Something is wrong with the formatting here, not sure exactly what. Here's my guess:

```suggestion
            i.e., `FOO = FOO ?: "default"`
```